### DISCUSSION TITLE
Checkout as Shopping Cart may cause UX issues

### DISCUSSION THREAD

**Discussion by tlosbar** [saleor/saleor#8154] [Q&A] [answered]
Hello, 

I've just seen the new doc for the incoming version 3.0 of Saleor [here](https://docs.saleor.io/docs/3.0/developer/checkout).
I quite agree with this principle but if I understand we have to create a Checkout before adding items to the "no cart"? 

So, what do you think about the unlogged users or anonymous users? 

Let me explain the problem in a little scenario.

An anonymous user is visiting the shop online. He can fetch the list of the products and consult the details of the desired product. It's ok, but if he wants to add this product to a cart, we have to redirect him to a login page before. There he can sign in, creates a new account, or just enter his email (because for anonymous user only an email is required) to pursue his goal of just adding a product to a cart.

If I have a good understanding of the new design, I think it can be a problem for the UX. 

Or can I have some help to understanding how I can manage a simple cart without having to ask the user anything on the shop before proceeding to a real checkout?

**Comment by maarcingebala** [comment, marked as answer]
Hey, one of the last features that we are missing before releasing the stable Saleor 3.0 is anonymous checkouts. Basically, you will be able to create a checkout without providing any email. It should allow handling cases as you described. We will start working on that very soon. Another option that is available already is to keep the cart in the browser and sync it with API once user decides to proceed to checkout and provide their email (this is currently implemented in our demo storefront).

**Reply by tlosbar** [reply]
Thanks. I'll check.
